File: frontend/next-generation-ticketing-system/src/Entity/CustomEntity/DriverRide.ts
import BusSchedule from "../BusSchedule";
import RouteDistrict from "../RouteDistrict";

class DriverRide {
  busSchedule: BusSchedule;
  routeDistricts: RouteDistrict[];

  constructor(busSchedule: BusSchedule, routeDistricts: RouteDistrict[]) {
    this.busSchedule = busSchedule;
    this.routeDistricts = routeDistricts;
  }
}

export default DriverRide;

File: backend/ngts/src/main/java/com/three/ngts/Entity/CustomEntity/DriverRide.java
package com.three.ngts.Entity.CustomEntity;

import com.three.ngts.Entity.BusSchedule;
import com.three.ngts.Entity.RouteDistrict;
import lombok.Data;

import java.util.List;

@Data
public class DriverRide {
    private BusSchedule busSchedule;
    private List<RouteDistrict> routeDistricts;

    public DriverRide(BusSchedule busSchedule, List<RouteDistrict> routeDistricts) {
        this.busSchedule = busSchedule;
        this.routeDistricts = routeDistricts;
    }
}
